atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/TracingKeyValueService.java

```diff
@@ -69,10 +69,10 @@ protected KeyValueService delegate() {
     }
 
     @Override
-    public void addGarbageCollectionSentinelValues(TableReference tableRef, Set<Cell> cells) {
+    public void addGarbageCollectionSentinelValues(TableReference tableRef, Iterable<Cell> cells) {
         //noinspection unused - try-with-resources closes trace
         try (CloseableTrace trace = startLocalTrace("addGarbageCollectionSentinelValues({}, {} cells)",
-                tableRef, cells.size())) {
+                tableRef, Iterables.size(cells))) {
```

Comment:
Most of the existing `Iterables.size` in `TracingKeyValueService` are actually called on un-filtered Collections so it does not end up being an issue and honestly I'm not that worried about this specific case for `addGarbageCollectionSentinelValues` as that's not really a hot path. If we see it burning CPU in profiles we can address later.